### ISSUE TITLE
source code?

### ISSUE THREAD

**Issue by jayanta525** [avafinger/nanopi-r2s-ubuntu-server-minimal-image#2]
where can i find the source for this ubuntu-build?

**Comment by shaneallgeier** [comment]
I agree, I would love to see the patches you applied to get everything working. Or are you just running a straight vanilla mainline kernel, @avafinger ?

(my goal is to get a completely up-to-date version of Arch Linux running on my R2S)

**Comment by Fail-Safe** [comment]
I have the same questions and interest as those above. Are you willing to make your source code available?

**Comment by shaneallgeier** [comment]
So I haven't had time to build things myself, but I suspect that @avafinger is basing his Ubuntu kernels off of the sources provided by FriendlyElec at this occasionally-updated Google drive folder:

https://drive.google.com/drive/folders/1OB4qN2Ehx8-CjoqdGv_ecPjQ1DgMnkf-

Be sure to come back and let us know if you're successful

**Comment by avafinger** [comment]
Kernel 5.8-rc is straight vanilla mainline with FE's USB3 work. i will provide the sources asap.

**Comment by Fail-Safe** [comment]
@jayanta525 I'm curious to see if the CPU cores are more balanced under load with Ubuntu than with OpenWrt on kernel 5.4.

**Comment by jayanta525** [comment]
@Fail-Safe with the irqbalance init.d script. I am seeing a pretty much-balanced load on each core.
